### PR TITLE
Enhancements to Traceable Integration

### PR DESCRIPTION
**Pull request by mtraceable**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Introduced new fields in the Incident to specify type of IP Address (ipAddressType) with values Internal or External
Introduced new fields in the Incident to specify type of API (apiType) with values Internal or External
Added new Incident field eventUrl containing the link to open the Incident Event in the Traceable Platform.
Ability to select the optional fields for the Incidents.
Ability to pull optional attributes of the affected APIs of the reported incidents:
Incidents for the selected HTTP Status Codes can now be ignored and not created in XSOAR.
Added additional input configuration for the Integration to provide the base url of the Traceable Platform UI endpoint.
Added additional input configuration to configure the HTTP Status Codes for which the Incidents should be ignored.
Updated the Docker image to: demisto/python3:3.10.12.68714.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by MichaelYochpaz — COMMENTED** [review]
Great job, looks great :)
Just a few minor notes.

File: Packs/Traceable/Integrations/Traceable/README.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ With this integration, an Incident can be raised in Cortex Xsoar when an Event i
 To use the integration the following mandatory parameters need to be set:
 |Parameter Name|Default Value|Description|
 |------|------|------|
```
[the remark this answers is not in the document]
Comment (mtraceable) [reply Packs/Traceable/Integrations/Traceable/README.md:10]: Hi @MichaelYochpaz do you mean something like:
`To use the integration the following mandatory parameters need to be set:

|Parameter Name|Default Value|Description|
|------|------|------|
|Traceable Platform API Endpoint URL|https://api.traceable.ai|Base URL of Traceable Platform API Endpoint.|
|`
I tried the fix you mentioned and if I do that, it does not format it correctly for me:
image
Instead I am placing a new line before the start of the table:
image
Let me know if this is ok.

[the remark this answers is not in the document]
Comment (mtraceable) [reply Packs/Traceable/Integrations/Traceable/README.md:11]: Done

**Comment by MichaelYochpaz** [comment]
@mtraceable Our internal build fails for:
```
/builds/xsoar/content/Packs/Traceable/Integrations/Traceable/Traceable.py:351:16: E9012 Client.__parse_status_code_strings: Demisto.log is found, Please replace all demisto.log usage with demisto.info or demisto.debug
/builds/xsoar/content/Packs/Traceable/Integrations/Traceable/Traceable.py:358:20: E9012 Client.__parse_status_code_strings: Demisto.log is found, Please replace all demisto.log usage with demisto.info or demisto.debug
/builds/xsoar/content/Packs/Traceable/Integrations/Traceable/Traceable.py:364:20: E9012 Client.__parse_status_code_strings: Demisto.log is found, Please replace all demisto.log usage with demisto.info or demisto.debug
/builds/xsoar/content/Packs/Traceable/Integrations/Traceable/Traceable.py:367:16: E9012 Client.__parse_status_code_strings: Demisto.log is found, Please replace all demisto.log usage with demisto.info or demisto.debug
/builds/xsoar/content/Packs/Traceable/Integrations/Traceable/Traceable.py:390:20: E9012 Client.__process_domain_event_field_list: Demisto.log is found, Please replace all demisto.log usage with demisto.info or demisto.debug
```
Please fix :)

**Comment by MichaelYochpaz** [comment]
Hey @mtraceable, PR seems to be good :)

Please see [this page](https://xsoar.pan.dev/docs/contributing/demo-prep), and message me in Slack (DFIR) when you're available for a demo so that I'll schedule a meeting.

**Review by MichaelYochpaz — APPROVED** [review]
@mtraceable Demo went great, good job! :)

File: Packs/Traceable/Integrations/Traceable/README.md [hunk 2]
```diff
@@ -29,6 +30,10 @@ The following parameters can be used to customize what Events should be exported
 |IP Reputation Level|Multi Select|No|CRITICAL, HIGH, MEDIUM|IP Reputations to query|
 |IP Abuse Velocity|Multi Select|No|CRITICAL, HIGH, MEDIUM|IP Abuse Velocity to query|
 |IP Location Type|Multi Select|No|-|IP Location Type to query|
+|Traceable Platform Endpoint URL|Long Text|No|https://app.traceable.ai|Base URL of the Traceable Platform UI Endpoint|
+|Ignore Status Codes|Long Text|No|400-499|Ignore Incidents for HTTP Requests failing with these Status Codes|
```
Comment (ShirleyDenkberg) [inline Packs/Traceable/Integrations/Traceable/README.md:34]:
```suggestion
|Ignore Status Codes|Long Text|No|400-499| Ignore incidents for HTTP requests failing with these status codes. |
```

File: Packs/Traceable/Integrations/Traceable/README.md [hunk 3]
```diff
@@ -29,6 +30,10 @@ The following parameters can be used to customize what Events should be exported
...
+|Incident optional field list|Multi Select|No|actorDevice,actorEntityId,actorId,actorScoreCategory,actorSession,anomalousAttribute,apiName,apiUri,category,ipAbuseVelocity,ipReputationLevel,securityEventType,securityScore,serviceId,serviceName,actorScore,threatCategory,type|Optional fields to pull from Traceable Event|
```
Comment (ShirleyDenkberg) [inline Packs/Traceable/Integrations/Traceable/README.md:35]:
```suggestion
|Incident optional field list|Multi select|No|actorDevice,actorEntityId,actorId,actorScoreCategory,actorSession,anomalousAttribute,apiName,apiUri,category,ipAbuseVelocity,ipReputationLevel,securityEventType,securityScore,serviceId,serviceName,actorScore,threatCategory,type|Optional fields to pull from Traceable Event|
```

File: Packs/Traceable/Integrations/Traceable/README.md [hunk 4]
```diff
@@ -29,6 +30,10 @@ The following parameters can be used to customize what Events should be exported
...
+|Additional API Attributes|Multi Select|No|isExternal,isAuthenticated,riskScore,riskScoreCategory,isLearnt|Additional API Attributes to query for the affected API in the Incident|
```
Comment (ShirleyDenkberg) [inline Packs/Traceable/Integrations/Traceable/README.md:36]:
```suggestion
|Additional API Attributes|Multi select|No|isExternal,isAuthenticated,riskScore,riskScoreCategory,isLearnt|Additional API Attributes to query for the affected API in the Incident|
```

File: Packs/Traceable/Integrations/Traceable/Traceable.yml [hunk 5]
```diff
@@ -99,11 +100,56 @@ configuration:
...
   required: false
   type: 13
+- display: Ignore Status Codes
+  name: ignoreStatusCodes
+  defaultvalue: 400-499
+  type: 0
+  required: false
+- display: Incident optional field list
+  name: optionalDomainEventFieldList
+  defaultvalue: actorDevice,actorEntityId,actorId,actorScoreCategory,actorSession,anomalousAttribute,apiName,apiUri,category,ipAbuseVelocity,ipReputationLevel,securityEventType,securityScore,serviceId,serviceName,actorScore,threatCategory,type
```
Comment (ShirleyDenkberg) [inline Packs/Traceable/Integrations/Traceable/Traceable.yml:120]: Do you need this (defaultvalue) if you have the list of options?

File: Packs/Traceable/Integrations/Traceable/Traceable.yml [hunk 6]
```diff
@@ -99,11 +100,56 @@ configuration:
...
+  - serviceId
+  - serviceName
+  - actorScore
+  - threatCategory
+  - type
+  type: 16
+  required: false
+- display: Additional API Attributes
+  name: optionalAPIAttributes
+  defaultvalue: isExternal,isAuthenticated,riskScore,riskScoreCategory,isLearnt
```
Comment (ShirleyDenkberg) [inline Packs/Traceable/Integrations/Traceable/Traceable.yml:144]: Do you need this (defaultvalue) if you have the list of options?

File: Packs/Traceable/ReleaseNotes/1_0_3.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
+
+#### Integrations
+##### Traceable
+- Updated the Docker image to: *demisto/python3:3.10.13.72123*.
+
+- Added new fields in the Incident to specify type of IP Address (**ipAddressType**) with values **Internal** or **External**.
```
Comment (ShirleyDenkberg) [inline Packs/Traceable/ReleaseNotes/1_0_3.md:6]:
```suggestion
- Added new fields in the incident to specify the type of IP address (**ipAddressType**) with values **Internal** or **External**.
```

File: Packs/Traceable/ReleaseNotes/1_0_3.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+- Added new fields in the Incident to specify type of API (**apiType**) with values **Internal** or **External**.
```
Comment (ShirleyDenkberg) [inline Packs/Traceable/ReleaseNotes/1_0_3.md:7]:
```suggestion
- Added new fields in the incident to specify the type of API (**apiType**) with values **Internal** or **External**.
```

File: Packs/Traceable/ReleaseNotes/1_0_3.md [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+- Added new Incident field **eventUrl** containing the link to open the Incident Event in the Traceable Platform.
```
Comment (ShirleyDenkberg) [inline Packs/Traceable/ReleaseNotes/1_0_3.md:8]:
```suggestion
- Added a new incident field **eventUrl** containing the link to open the incident event in the Traceable platform.
```

File: Packs/Traceable/ReleaseNotes/1_0_3.md [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+- Added ability to select the optional fields for the Incidents.
```
Comment (ShirleyDenkberg) [inline Packs/Traceable/ReleaseNotes/1_0_3.md:9]:
```suggestion
- Added the ability to select the optional fields for the incidents.
```

File: Packs/Traceable/ReleaseNotes/1_0_3.md [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+  - ipAbuseVelocity
+  - ipReputationLevel
+  - securityEventType
+  - securityScore
+  - serviceId
+  - serviceName
+  - actorScore
+  - threatCategory
+  - type
+- Added ability to pull optional attributes of the affected APIs of the reported incidents.
```
Comment (ShirleyDenkberg) [inline Packs/Traceable/ReleaseNotes/1_0_3.md:28]:
```suggestion
- Added the ability to pull optional attributes of the affected APIs of the reported incidents.
```

File: Packs/Traceable/ReleaseNotes/1_0_3.md [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+  - isExternal
+  - isAuthenticated
+  - riskScore
+  - riskScoreCategory
+  - isLearnt
+- Added an option to ignore incidents failing with specific HTTP Status Codes set on instance's configuration.
```
Comment (ShirleyDenkberg) [inline Packs/Traceable/ReleaseNotes/1_0_3.md:34]:
```suggestion
- Added an option to ignore incidents failing with specific HTTP status codes set on the instance's configuration.
```

File: Packs/Traceable/ReleaseNotes/1_0_3.md [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+- Added additional input configuration for the Integration to provide the base url of the Traceable Platform UI endpoint.
```
Comment (ShirleyDenkberg) [inline Packs/Traceable/ReleaseNotes/1_0_3.md:35]:
```suggestion
- Added an additional input configuration for the integration to provide the base URL of the Traceable platform UI endpoint.
```

**Comment by ShirleyDenkberg** [comment]
@MichaelYochpaz Doc review completed.

File: Packs/Traceable/Integrations/Traceable/Traceable.yml (hunk 5 above)
Comment (mtraceable) [reply Packs/Traceable/Integrations/Traceable/Traceable.yml:120]: Hi @ShirleyDenkberg the idea was to have it fetch as many fields to start with, and the customer can take a conscious decision to remove or retain the ones that they wish.

File: Packs/Traceable/Integrations/Traceable/Traceable.yml (hunk 6 above)
Comment (mtraceable) [reply Packs/Traceable/Integrations/Traceable/Traceable.yml:144]: @ShirleyDenkberg same case here - the idea was to have it fetch as many fields to start with, and the customer can take a conscious decision to remove or retain the ones that they wish.

**Pull request merged**